Core/HLE/sceNet.cpp

```diff
@@ -43,6 +50,168 @@ static struct SceNetMallocStat netMallocStat;
 
 static std::map<int, ApctlHandler> apctlHandlers;
 
+#ifdef _WIN32
+static HANDLE hIDMapFile = NULL;
+#elif (__linux__ && !defined(ANDROID)) || __APPLE__
+static int hIDMapFile = 0;
+#endif
+static int32_t* pIDBuf = NULL;
+#define ID_SHM_NAME "PPSSPP_ID"
+
+// Get current number of instance of PPSSPP running.
+static uint8_t getInstanceNumber() {
+#ifdef _WIN32
+#if defined(_XBOX)
+	uint32_t BUF_SIZE = 0x10000; // 64k in 360;
+#else
+	uint32_t BUF_SIZE = 4096;
+	SYSTEM_INFO sysInfo;
+
+	GetSystemInfo(&sysInfo);
+	int gran = sysInfo.dwAllocationGranularity ? sysInfo.dwAllocationGranularity : 0x10000;
+	BUF_SIZE = (BUF_SIZE + gran - 1) & ~(gran - 1);
+#endif
+
+	hIDMapFile = CreateFileMapping(
+		INVALID_HANDLE_VALUE,    // use paging file
+		NULL,                    // default security
+		PAGE_READWRITE,          // read/write access
+		0,                       // maximum object size (high-order DWORD)
+		BUF_SIZE,                // maximum object size (low-order DWORD)
+		TEXT(ID_SHM_NAME));       // name of mapping object
+
+	DWORD lasterr = GetLastError();
+	if (hIDMapFile == NULL)
+	{
+		ERROR_LOG(SCENET, "Could not create %s file mapping object (%d).", ID_SHM_NAME, lasterr);
+		return 1;
+	}
+	pIDBuf = (int32_t*)MapViewOfFile(hIDMapFile,   // handle to map object
+		FILE_MAP_ALL_ACCESS, // read/write permission
+		0,
+		0,
+		sizeof(int32_t)); //BUF_SIZE
+
+	if (pIDBuf == NULL)
+	{
+		ERROR_LOG(SCENET, "Could not map view of file %s (%d).", ID_SHM_NAME, GetLastError());
+		//CloseHandle(hIDMapFile);
+		return 1;
+	}
+
+	(*pIDBuf) = std::max(1, ((*pIDBuf) + 1) % 256);
+	int id = *pIDBuf;
+	UnmapViewOfFile(pIDBuf);
+	//CloseHandle(hIDMapFile); //Should be called when program exits
+	//hIDMapFile = NULL;
+
+	return id;
+#elif (__linux__ && !defined(ANDROID)) || __APPLE__
+	long BUF_SIZE = 4096;
+	//caddr_t pIDBuf;
+	int status;
+
+	// Create shared memory object 
+
+	hIDMapFile = shm_open(ID_SHM_NAME, O_CREAT | O_RDWR, 0);
```

Comment:
Yes, and currently it's like that, since CloseHandle(hIDMapFile); / shm_unlink(ID_SHM_NAME); which is responsible to reset the number are being commented.